src/bridge/cockpitcpusamples.c

```diff
@@ -188,12 +205,22 @@ cockpit_cpu_temperature (CockpitSamples *samples)
       g_strchomp (name);
 
       // compare device name with CPU names
-      // Intel: coretemp, AMD: k8temp or k10temp, ARM: cpu_thermal
+      // Intel: coretemp, AMD: k8temp or k10temp, ARM: cpu_thermal, Asus motherboard: atk0110
       if (g_str_equal (name, "coretemp") || g_str_equal (name, "cpu_thermal") ||
-          g_str_equal (name, "k8temp") || g_str_equal (name, "k10temp"))
+          g_str_equal (name, "k8temp") || g_str_equal (name, "k10temp") || g_str_equal (name, "atk0110"))
         {
           // hwmon contains CPU info
-          sample_cpu_sensors (samples, i);
+          detect_cpu_sensors (devices, name, i);
         }
     }
 }
+
+void
+cockpit_cpu_temperature (CockpitSamples *samples)
+{
+  static GList *devices = NULL;
+  if (devices == NULL)
+      detect_hwmon_device (&devices);
```

Comment:
This isn't sufficient -- if this does not find any sensors (like in a VM or embedded device), this will run the initialization over and over again. Please introduce a separate `static gboolean initialized = FALSE` and use that to run (or not) the detection. I.e. you need to tell apart "never run" from "nothing found".

This will also "break" the tests with the sleep.

Reply:
Thanks for pointing this out!